src/pieces/counting.ts

```diff
@@ -0,0 +1,127 @@
+import { Client, MessageEmbed, TextChannel } from 'discord.js';
+import { DB, CHANNELS } from '@root/config';
+import { Counter } from '../lib/types/Counter';
+
+let countingDB: Counter;
+let countingChannel;
+
+async function register(bot: Client): Promise<void> {
+	countingDB = await bot.mongo.collection(DB.COUNTING).findOne({ piece: 'counting' });
+	countingChannel = bot.channels.cache.get(CHANNELS.COUNTING_CHANNEL) as TextChannel;
+
+	bot.on('messageUpdate', async msg => {
+		if (msg.channel.id !== CHANNELS.COUNTING_CHANNEL) return;
+		if (msg.author.bot && !msg.partial) return; // sage will crash if it hits this line and the message is a partial
+		if (msg.createdTimestamp <= countingDB.startDate) return;
+
+		const endMsg = `${msg.partial ? 'Someone' : msg.author.username} edited a message and broke the count!`;
+
+		if (countingDB.count > 0) {
+			await endGame(endMsg, bot);
+		}
+	});
+
+	bot.on('messageDelete', async msg => {
+		if (msg.channel.id !== CHANNELS.COUNTING_CHANNEL) return;
+		if (msg.author.bot && !msg.partial) return; // sage will crash if it hits this line and the message is a partial
+		if (msg.createdTimestamp <= countingDB.startDate) return;
+
+		const endMsg = `${msg.partial ? 'Someone' : msg.author.username} deleted a message and broke the count!`;
+
+		if (countingDB.count > 0) {
+			await endGame(endMsg, bot);
+		}
+	});
+
+	bot.on('messageCreate', async msg => {
+		if (msg.author.bot) return;
+
+		if (msg.channel.id === CHANNELS.COUNTING_CHANNEL) {
+			let gameOver = false;
+
+			if (msg.author.id === countingDB.lastSendId) {
+				gameOver = true;
+				msg.reply(`${msg.author.username}, you aren't allowed to count twice in a row!`);
+			}
+
+			countingDB.lastSendId = msg.author.id;
+
+			let endMsg = '';
+			if (msg.stickers.size > 0) {
+				gameOver = true;
+				endMsg = `${msg.author.username}, that's a nice sticker mate but we're supposed to be counting here.`;
+			} else if (msg.attachments.size > 0) {
+				gameOver = true;
+				endMsg = `${msg.author.username}, you can't send attachments here. This is a counting channel.`;
+			} else if (Number.isNaN(parseInt(msg.content))) {
+				gameOver = true;
+				endMsg = `Really ${msg.author.username}? ${msg.content} isn't a number!`;
+			} else if (parseInt(msg.content) > countingDB.count + 1) {
+				gameOver = true;
+				endMsg = `Come on ${msg.author.username}, ${msg.content} doesn't come after **${countingDB.count}**!`;
+			} else if (parseInt(msg.content) < countingDB.count) {
+				gameOver = true;
+				endMsg = `Believe it or not ${msg.author.username}, ${msg.content} is less than **${countingDB.count}**.`;
+			} else if (parseInt(msg.content) === countingDB.count) {
+				gameOver = true;
+				endMsg = `My dude that is literally the number we were on.`;
+			}
```

Comment:
This else if chain doesn't account for someone sending a message that starts with a number but has more after it. For instance `42 is the answer to the ultimate question of life, the universe and everything` will fall through this check and then fail the one below putting the channel into a broken state.